### PR TITLE
Tests for scaled content with masks and mask-type property

### PR DESCRIPTION
**Pull request by gsnedders**
---

Originally posted as https://github.com/w3c/csswg-test/pull/619 by @dirkschulze on 22 Oct 2014, 07:13 UTC:

>

### CODE REVIEW

**Comment by csswg-test-issue-mover** [comment]
Originally posted as https://github.com/w3c/csswg-test/pull/619#issuecomment-60045107 by @syncbot on 22 Oct 2014, 07:13 UTC:

> ### Automatic validation checks of commit 428bfd3 passed.

**Comment by csswg-test-issue-mover** [comment]
Originally posted as https://github.com/w3c/csswg-test/pull/619#issuecomment-60199299 by @syncbot on 23 Oct 2014, 06:52 UTC:

> ### Automatic validation checks of commit 92c3a80 passed.

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/217155001)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 43d6601be1da376113490c1f69a59cd537758802
Using browser at version BuildID 20170331102157; SourceStamp 8df9fabf2587b7020889755acb9e75b664fe13cf
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>6 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5279/css/css-masking-1/mask-svg-content/mask-negative-scale.svg">/css/css-masking-1/mask-svg-content/mask-negative-scale.svg</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5279/css/css-masking-1/mask-svg-content/mask-text-001.svg">/css/css-masking-1/mask-svg-content/mask-text-001.svg</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5279/css/css-masking-1/mask-svg-content/mask-type-001.svg">/css/css-masking-1/mask-svg-content/mask-type-001.svg</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5279/css/css-masking-1/mask-svg-content/mask-type-002.svg">/css/css-masking-1/mask-svg-content/mask-type-002.svg</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5279/css/css-masking-1/mask-svg-content/mask-type-003.svg">/css/css-masking-1/mask-svg-content/mask-type-003.svg</a></summary>


| Subtest | Results | Messages |
|---------|---------|----------|
|         | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5279/innerText/setter.html">/innerText/setter.html</a></summary>


|                                        Subtest                                        | Results | Messages |
|---------------------------------------------------------------------------------------|---------|----------|
|                                                                                       | OK      |          |
| `Simplest possible test`                                                              | PASS    |          |
| `Simplest possible test, detached`                                                    | PASS    |          |
| `Newlines convert to <br> in non-white-space:pre elements`                            | PASS    |          |
| `Newlines convert to <br> in non-white-space:pre elements, detached`                  | PASS    |          |
| `Newlines convert to <br> in <pre> element`                                           | PASS    |          |
| `Newlines convert to <br> in <pre> element, detached`                                 | PASS    |          |
| `Newlines convert to <br> in <textarea> element`                                      | PASS    |          |
| `Newlines convert to <br> in <textarea> element, detached`                            | PASS    |          |
| `Newlines convert to <br> in white-space:pre element`                                 | PASS    |          |
| `Newlines convert to <br> in white-space:pre element, detached`                       | PASS    |          |
| `CRs convert to <br> in non-white-space:pre elements`                                 | PASS    |          |
| `CRs convert to <br> in non-white-space:pre elements, detached`                       | PASS    |          |
| `CRs convert to <br> in <pre> element`                                                | PASS    |          |
| `CRs convert to <br> in <pre> element, detached`                                      | PASS    |          |
| `Newline/CR pair converts to <br> in non-white-space:pre element`                     | PASS    |          |
| `Newline/CR pair converts to <br> in non-white-space:pre element, detached`           | PASS    |          |
| `Newline/newline pair converts to two <br>s in non-white-space:pre element`           | PASS    |          |
| `Newline/newline pair converts to two <br>s in non-white-space:pre element, detached` | PASS    |          |
| `CR/CR pair converts to two <br>s in non-white-space:pre element`                     | PASS    |          |
| `CR/CR pair converts to two <br>s in non-white-space:pre element, detached`           | PASS    |          |
| `CRs convert to <br> in white-space:pre element`                                      | PASS    |          |
| `CRs convert to <br> in white-space:pre element, detached`                            | PASS    |          |
| `< preserved`                                                                         | PASS    |          |
| `< preserved, detached`                                                               | PASS    |          |
| `> preserved`                                                                         | PASS    |          |
| `> preserved, detached`                                                               | PASS    |          |
| `& preserved`                                                                         | PASS    |          |
| `& preserved, detached`                                                               | PASS    |          |
| `" preserved`                                                                         | PASS    |          |
| `" preserved, detached`                                                               | PASS    |          |
| `' preserved`                                                                         | PASS    |          |
| `' preserved, detached`                                                               | PASS    |          |
| `innerText not supported on SVG elements`                                             | PASS    |          |
| `innerText not supported on SVG elements, detached`                                   | PASS    |          |
| `innerText not supported on MathML elements`                                          | PASS    |          |
| `innerText not supported on MathML elements, detached`                                | PASS    |          |
| `Null characters preserved`                                                           | PASS    |          |
| `Null characters preserved, detached`                                                 | PASS    |          |
| `Tabs preserved`                                                                      | PASS    |          |
| `Tabs preserved, detached`                                                            | PASS    |          |
| `Leading whitespace preserved`                                                        | PASS    |          |
| `Leading whitespace preserved, detached`                                              | PASS    |          |
| `Trailing whitespace preserved`                                                       | PASS    |          |
| `Trailing whitespace preserved, detached`                                             | PASS    |          |
| `Whitespace not compressed`                                                           | PASS    |          |
| `Whitespace not compressed, detached`                                                 | PASS    |          |
| `Existing text deleted`                                                               | PASS    |          |
| `Existing text deleted, detached`                                                     | PASS    |          |
| `Existing <br> deleted`                                                               | PASS    |          |
| `Existing <br> deleted, detached`                                                     | PASS    |          |
| `Assigning the empty string`                                                          | PASS    |          |
| `Assigning the empty string, detached`                                                | PASS    |          |
| `Assigning null`                                                                      | PASS    |          |
| `Assigning null, detached`                                                            | PASS    |          |
| `Assigning undefined`                                                                 | PASS    |          |
| `Assigning undefined, detached`                                                       | PASS    |          |
| `Start with CR`                                                                       | PASS    |          |
| `Start with CR, detached`                                                             | PASS    |          |
| `Start with LF`                                                                       | PASS    |          |
| `Start with LF, detached`                                                             | PASS    |          |
| `Start with CRLF`                                                                     | PASS    |          |
| `Start with CRLF, detached`                                                           | PASS    |          |
| `End with CR`                                                                         | PASS    |          |
| `End with CR, detached`                                                               | PASS    |          |
| `End with LF`                                                                         | PASS    |          |
| `End with LF, detached`                                                               | PASS    |          |
| `End with CRLF`                                                                       | PASS    |          |
| `End with CRLF, detached`                                                             | PASS    |          |
| `innerText on <area> element`                                                         | PASS    |          |
| `innerText on <area> element, detached`                                               | PASS    |          |
| `innerText on <base> element`                                                         | PASS    |          |
| `innerText on <base> element, detached`                                               | PASS    |          |
| `innerText on <basefont> element`                                                     | PASS    |          |
| `innerText on <basefont> element, detached`                                           | PASS    |          |
| `innerText on <bgsound> element`                                                      | PASS    |          |
| `innerText on <bgsound> element, detached`                                            | PASS    |          |
| `innerText on <br> element`                                                           | PASS    |          |
| `innerText on <br> element, detached`                                                 | PASS    |          |
| `innerText on <col> element`                                                          | PASS    |          |
| `innerText on <col> element, detached`                                                | PASS    |          |
| `innerText on <embed> element`                                                        | PASS    |          |
| `innerText on <embed> element, detached`                                              | PASS    |          |
| `innerText on <frame> element`                                                        | PASS    |          |
| `innerText on <frame> element, detached`                                              | PASS    |          |
| `innerText on <hr> element`                                                           | PASS    |          |
| `innerText on <hr> element, detached`                                                 | PASS    |          |
| `innerText on <image> element`                                                        | PASS    |          |
| `innerText on <image> element, detached`                                              | PASS    |          |
| `innerText on  element`                                                          | PASS    |          |
| `innerText on  element, detached`                                                | PASS    |          |
| `innerText on <input> element`                                                        | PASS    |          |
| `innerText on <input> element, detached`                                              | PASS    |          |
| `innerText on <keygen> element`                                                       | PASS    |          |
| `innerText on <keygen> element, detached`                                             | PASS    |          |
| `innerText on <link> element`                                                         | PASS    |          |
| `innerText on <link> element, detached`                                               | PASS    |          |
| `innerText on <menuitem> element`                                                     | PASS    |          |
| `innerText on <menuitem> element, detached`                                           | PASS    |          |
| `innerText on <meta> element`                                                         | PASS    |          |
| `innerText on <meta> element, detached`                                               | PASS    |          |
| `innerText on <param> element`                                                        | PASS    |          |
| `innerText on <param> element, detached`                                              | PASS    |          |
| `innerText on <source> element`                                                       | PASS    |          |
| `innerText on <source> element, detached`                                             | PASS    |          |
| `innerText on <track> element`                                                        | PASS    |          |
| `innerText on <track> element, detached`                                              | PASS    |          |
| `innerText on <wbr> element`                                                          | PASS    |          |
| `innerText on <wbr> element, detached`                                                | PASS    |          |
| `innerText on <colgroup> element`                                                     | PASS    |          |
| `innerText on <colgroup> element, detached`                                           | PASS    |          |
| `innerText on <frameset> element`                                                     | PASS    |          |
| `innerText on <frameset> element, detached`                                           | PASS    |          |
| `innerText on <head> element`                                                         | PASS    |          |
| `innerText on <head> element, detached`                                               | PASS    |          |
| `innerText on <html> element`                                                         | PASS    |          |
| `innerText on <html> element, detached`                                               | PASS    |          |
| `innerText on <table> element`                                                        | PASS    |          |
| `innerText on <table> element, detached`                                              | PASS    |          |
| `innerText on <tbody> element`                                                        | PASS    |          |
| `innerText on <tbody> element, detached`                                              | PASS    |          |
| `innerText on <tfoot> element`                                                        | PASS    |          |
| `innerText on <tfoot> element, detached`                                              | PASS    |          |
| `innerText on <thead> element`                                                        | PASS    |          |
| `innerText on <thead> element, detached`                                              | PASS    |          |
| `innerText on <tr> element`                                                           | PASS    |          |
| `innerText on <tr> element, detached`                                                 | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/217155002)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 43d6601be1da376113490c1f69a59cd537758802
Using browser at version 59.0.3053.3 dev
Starting 10 test iterations
## Unstable results ##

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5279/)

**Comment by foolip** [comment]
@gsnedders can you rebase this so we get a new Travis run? I can review if nobody else shows up.

**Comment by dirkschulze** [comment]
@foolip that would be great. Not sure if I should review the tests when I wrote them :P (Though I can even remember writing the tests.)

**Comment by foolip** [comment]
w3c-test:mirror

File: css/css-masking/mask-svg-content/mask-type-001.svg [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:html="http://www.w3.org/1999/xhtml">
+<g id="testmeta">
+	<title>CSS Masking: mask without mask-type alpha</title>
+	<html:link rel="author" title="Dirk Schulze" href="mailto:dschulze@adobe.com"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#svg-masks"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#MaskElement"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#the-mask-type"/>
+	<html:link rel="match" href="reference/mask-text-001-ref.svg"/>
```
Comment (foolip) [inline css/css-masking/mask-svg-content/mask-type-001.svg]: Wrong ref, this will fail.

File: css/css-masking/mask-svg-content/mask-type-002.svg [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:html="http://www.w3.org/1999/xhtml">
+<g id="testmeta">
+	<title>CSS Masking: mask without mask-type luminance</title>
+	<html:link rel="author" title="Dirk Schulze" href="mailto:dschulze@adobe.com"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#svg-masks"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#MaskElement"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#the-mask-type"/>
+	<html:link rel="match" href="reference/mask-text-001-ref.svg"/>
```
Comment (foolip) [inline css/css-masking/mask-svg-content/mask-type-002.svg]: Also the wrong ref

File: css/css-masking/mask-svg-content/mask-type-003.svg [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:html="http://www.w3.org/1999/xhtml">
+<g id="testmeta">
+	<title>CSS Masking: mask without specified mask-type</title>
+	<html:link rel="author" title="Dirk Schulze" href="mailto:dschulze@adobe.com"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#svg-masks"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#MaskElement"/>
+	<html:link rel="help" href="http://www.w3.org/TR/css-masking-1/#the-mask-type"/>
+	<html:link rel="match" href="reference/mask-text-001-ref.svg"/>
```
Comment (foolip) [inline css/css-masking/mask-svg-content/mask-type-003.svg]: Wrong ref

File: css/css-masking/mask-svg-content/reference/mask-green-square-001-ref.svg [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:html="http://www.w3.org/1999/xhtml">
+<g id="testmeta">
+	<title>CSS Masking: Reftest reference</title>
+	<html:link rel="author" title="Dirk Schulze" href="mailto:dschulze@adobe.com"/>
+	<metadata class="flags">svg</metadata>
+</g>
+<rect x="50" y="50" width="100" height="100" fill="green"/>
```
Comment (foolip) [inline css/css-masking/mask-svg-content/reference/mask-green-square-001-ref.svg]: This ref is unused.

**Review by foolip — APPROVED** [review]
LGTM % wrong refs. If you could `wpt run` this on Chrome and/or Firefox to check for other bogus failure causes, that'd be great.

**Comment by foolip** [comment]
@gsnedders I think so yes. That someone would be you since you created the PR :)

**Comment by gsnedders** [comment]
@foolip only because I copied all the open csswg-test PRs over when the repos merged; I certainly don't want to take ownership of all of them!

File: css/css-masking/mask-svg-content/reference/mask-green-square-001-ref.svg (hunk 4 above)
Comment (zcorpan) [reply css/css-masking/mask-svg-content/reference/mask-green-square-001-ref.svg]: It should be used by mask-type-*

**Comment by zcorpan** [comment]
I've addressed the comments @foolip @gsnedders

File: css/css-masking/mask-svg-content/reference/mask-green-square-001-ref.svg (hunk 4 above)
Comment (foolip) [reply css/css-masking/mask-svg-content/reference/mask-green-square-001-ref.svg]: And now it is.

**Review by foolip — APPROVED** [review]

**Pull request merged**